Panic when calling an exported method incorrectly

I need to implement a method that fills strings passed in by a caller:

    <method name="GetServerInformation">
      <arg name="return_name" type="s" direction="out"/>
      <arg name="return_vendor" type="s" direction="out"/>
      <arg name="return_version" type="s" direction="out"/>
      <arg name="return_spec_version" type="s" direction="out"/>
    </method>

My approach at this was to export a method with a signature of (s1, s2, s3, s4 *string) *dbus.Error, but I don't think this is correct -  that isn't the point of this report, though. I accidentally called the method with four regular strings, and subsequently caused a crash:

```
method call time=1585891966.401733 sender=:1.5810 -> destination=:1.5811 serial=389 path=/org/freedesktop/Notifications; interface=org.freedesktop.Notifications; member=GetServerInformation
   string "a"
   string "a"
   string "a"
   string "a"
signal time=1585891966.405256 sender=org.freedesktop.DBus -> destination=:1.5811 serial=4294967295 path=/org/freedesktop/DBus; interface=org.freedesktop.DBus; member=NameLost
```

I guess this isn't high risk, but it might still be an idea to try to be robust against bad input. Backtrace follows.

```
panic: reflect: call of reflect.Value.Type on zero Value

goroutine 39 [running]:
reflect.Value.Type(0x0, 0x0, 0x0, 0x67c800, 0x5e39c0)
	/usr/lib/golang/src/reflect/value.go:1877 +0x166
github.com/godbus/dbus.setDest(0x0, 0x0, 0x0, 0x5e39c0, 0xc000129520, 0x98, 0x5dc400, 0x5dc400)
	/home/burchr/code/go/src/github.com/godbus/dbus/dbus.go:83 +0x632
github.com/godbus/dbus.storeBase(0x0, 0x0, 0x0, 0x5e39c0, 0xc000129520, 0x98, 0x7cc310, 0xc0001444e0)
	/home/burchr/code/go/src/github.com/godbus/dbus/dbus.go:79 +0x5d
github.com/godbus/dbus.store(0x0, 0x0, 0x0, 0x5e39c0, 0xc000129520, 0x98, 0x5f32c0, 0x0)
	/home/burchr/code/go/src/github.com/godbus/dbus/dbus.go:74 +0xe6
github.com/godbus/dbus.store(0x5dc400, 0xc000140020, 0x196, 0x5e39c0, 0xc000129520, 0x98, 0xc00014ac90, 0x43eb83)
	/home/burchr/code/go/src/github.com/godbus/dbus/dbus.go:66 +0x1c1
github.com/godbus/dbus.store(0xc000132400, 0xc000140020, 0x16, 0x5e39c0, 0xc000129520, 0x98, 0x4, 0x2)
	/home/burchr/code/go/src/github.com/godbus/dbus/dbus.go:66 +0x1c1
github.com/godbus/dbus.storeInterfaces(0x5e39c0, 0xc000129520, 0xc000132400, 0xc000140020, 0x7ebf18, 0xa93)
	/home/burchr/code/go/src/github.com/godbus/dbus/dbus.go:61 +0x123
github.com/godbus/dbus.Store(0xc000132340, 0x4, 0x4, 0xc0001323c0, 0x4, 0x4, 0xc0000555b0, 0x0)
	/home/burchr/code/go/src/github.com/godbus/dbus/dbus.go:53 +0x88
github.com/godbus/dbus.standardMethodArgumentDecode(0x678620, 0xc0000db480, 0xc000125780, 0x7, 0xc0001238f0, 0xc000132340, 0x4, 0x4, 0xc0000e6660, 0xc00011ea40, ...)
	/home/burchr/code/go/src/github.com/godbus/dbus/export.go:93 +0x5fd
github.com/godbus/dbus.(*Conn).decodeArguments(0xc0000c6000, 0x678620, 0xc0000db480, 0xc000125780, 0x7, 0xc0001238f0, 0x7ebf18, 0x0, 0x0, 0x0, ...)
	/home/burchr/code/go/src/github.com/godbus/dbus/export.go:104 +0x161
github.com/godbus/dbus.(*Conn).handleCall(0xc0000c6000, 0xc0001238f0)
	/home/burchr/code/go/src/github.com/godbus/dbus/export.go:147 +0x2e5
created by github.com/godbus/dbus.(*Conn).inWorker
```

--- a/dbus.go
+++ b/dbus.go
@@ -63,6 +63,9 @@ func storeInterfaces(src, dest interface{}) error {
 
 func store(dest, src reflect.Value) error {
 	if dest.Kind() == reflect.Ptr {
+		if dest.IsNil() {
+			dest.Set(reflect.New(dest.Type().Elem()))
+		}
 		return store(dest.Elem(), src)
 	}
 	switch src.Kind() {

